manticore/core/manticore.py

```diff
@@ -1139,3 +1178,41 @@ def save_run_data(self):
             config.save(f)
 
         logger.info("Results in %s", self._output.store.uri)
+
+    @at_not_running
+    def set_instrospection_plugin(self, new_plugin: type):
+        """
+        Allows the user to extend the functionality of the default IntrospectionAPI plugin.
+        Must be called before ManticoreBase.run()
+
+        :param new_plugin: a subclass of IntrospectionAPIPlugin
+        """
+        assert issubclass(
+            new_plugin, self._introspection_plugin
+        ), "New plugin must be a subclass of IntrospectionAPIPlugin"
+        self._introspection_plugin = new_plugin
```

Comment:
We have the reference to this plugin from outside manticore. Why do we need t keep a specific name for it if it is going to be invoked via events.

Reply:
All the internal data is updated by events, but there's one external method - `get_state_descriptors` - that's called by `ManticoreBase.introspect`. 

This particular function exists so that a user can change the type of `_introspector` before instantiating it - a feature @ekilmer requested. I wanted to avoid adding too many parameters to `__init__`, so I made it a field on the object with a function to change it. Frankly though, I'm starting to think it might be more elegant to bite the bullet and add a parameter. Thoughts?